conda/configure_compass_env.py

```diff
@@ -0,0 +1,946 @@
+            os.path.join(esmf_path, 'bin')))
+        sys_info['env_vars'].append(
+            'export LD_LIBRARY_PATH={}:$LD_LIBRARY_PATH'.format(
+                os.path.join(esmf_path, 'lib')))
+
+    if scorpio != 'None':
+        sys_info['env_vars'].append('export PIO={}'.format(scorpio_path))
+
+    build_esmf = 'False'
+    if esmf == 'None':
+        esmf_branch = 'None'
+    else:
+        esmf_branch = 'ESMF_{}'.format(esmf.replace('.', '_'))
+        if not os.path.exists(esmf_path) or force_build:
+            build_esmf = 'True'
+
+    build_scorpio = 'False'
+    if scorpio != 'None' and (not os.path.exists(scorpio_path) or force_build):
+        build_scorpio = 'True'
+
+    script_filename = 'build.bash'
+
+    with open('{}/build.template'.format(template_path), 'r') as f:
+        template = Template(f.read())
+
+    modules = '\n'.join(sys_info['modules'])
+
+    if machine is None:
+        # need to activate the conda environment because that's where the
+        # libraries are
+        modules = '{}\n{}'.format(activate_env.replace('; ', '\n'), modules)
+
+    script = template.render(
+        sys_info=sys_info, modules=modules,
+        scorpio=scorpio, scorpio_path=scorpio_path,
+        build_scorpio=build_scorpio, esmf_path=esmf_path,
+        esmf_branch=esmf_branch, build_esmf=build_esmf)
+    print('Writing {}'.format(script_filename))
+    with open(script_filename, 'w') as handle:
+        handle.write(script)
+
+    command = '/bin/bash build.bash'
+    check_call(command)
+
+    return sys_info, system_libs
+
+
+def write_load_compass(template_path, activ_path, conda_base, is_test, version,
+                       activ_suffix, prefix, env_name, machine, sys_info,
+                       env_only):
+
+    try:
+        os.makedirs(activ_path)
+    except FileExistsError:
+        pass
+
+    if is_test:
+        if prefix is None:
+            prefix = 'test_compass_{}'.format(version)
+    else:
+        prefix = 'load_compass_{}'.format(version)
+
+    script_filename = '{}/{}{}.sh'.format(activ_path, prefix, activ_suffix)
+
+    if not env_only:
+        sys_info['env_vars'].append('export USE_PIO2=true')
+    sys_info['env_vars'].append('export HDF5_USE_FILE_LOCKING=FALSE')
+    sys_info['env_vars'].append('export LOAD_COMPASS_ENV={}'.format(
+        script_filename))
+    if machine is not None:
+        sys_info['env_vars'].append('export COMPASS_MACHINE={}'.format(
+            machine))
+
+    filename = '{}/load_compass.template'.format(template_path)
+    with open(filename, 'r') as f:
+        template = Template(f.read())
+
+    if is_test:
+        update_compass = \
+            'if [[ -f "./setup.py" && -d "compass" ]]; then\n' \
+            '   # safe to assume we\'re in the compass repo\n' \
+            '   # update the compass installation to point here\n' \
+            '   python -m pip install -e .\n' \
+            'fi'
```

Comment:
Okay, this worked well in a test on my laptop.  I think it's pretty harmless and will hopefully help to prevent confusion when you source the activation script in a new repo.  It's also fast enough to not slow down anyone's workflow.